File: solutions/golang/1/solution.go
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, v := range nums {
		map[v] = i
	}

	var ans = []int{0, 0}
	for i, v := range(nums) {
		diff := target - v
		location, ok := m[diff]
		if ok && location != i {
			ans[0] = i
			ans[1] = location
			break
		}
	}

	return ans
}

File: solutions/cpp/1/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target) {
        unordered_map<int, int> map;
        for (int i = 0; i < nums.size(); i++) {
            map[nums[i]] = i;
        }

        vector<int> ans;
        for (int i = 0; i < nums.size(); i++) {
            if (map.find(target - nums[i]) != map.end()) {
                if (map.find(target - nums[i]) != i) {
                    ans.push_back(i);
                    ans.push_back(map[target - nums[i]]);
                    break;
                }
            }
        }
        return ans;
    }
}